src/infra/src/table/alerts/mod.rs

```diff
@@ -283,6 +301,21 @@ pub async fn list<C: ConnectionTrait>(
     Ok(alerts)
 }
 
+/// Tries to get an alert ORM entity and its parent folder ORM entity.
+async fn get_model_by_id<C: ConnectionTrait>(
+    conn: &C,
+    org_id: &str,
+    alert_id: Ksuid,
+) -> Result<Option<(folders::Model, alerts::Model)>, sea_orm::DbErr> {
+    let maybe_f_a = alerts::Entity::find_by_id(alert_id.to_string())
+        .filter(alerts::Column::Org.eq(org_id))
+        .find_also_related(folders::Entity)
+        .one(conn)
+        .await?
+        .and_then(|(a, maybe_f)| maybe_f.map(|f| (f, a)));
```

Comment:
Just to understand: looks like folder is guaranteed to exist if alert with id does, as in it wouldn't be the case where it finds the alert by its id, but folder is none, which return both to none?

Reply:
That's correct. `folder_id` is a non-nullable foreign key on the `alerts` table so if an alert record exists then it is guaranteed to have a parent folder which also exists.